### PR TITLE
Docs: Adds Required Imports and Initialization section in user_guide

### PR DESCRIPTION
**Pull request by naxatra2**
Adds #972
I have added the required import calls, in each code blocks that I found necessary. 

I was unsure about the prerequisite setups and model initializations, so I did not include much about them.

### CODE REVIEW

**Comment by bw4sz** [comment]
I am ambivalent here. Maybe @henrykironde or @ethanwhite have an opinion whether each code block inside the doc is on its own reproducible.

File: docs/user_guide/09_configuration_file.md [hunk 1]
```diff
@@ -104,7 +106,7 @@ Most commonly, `cpu`, `gpu` or `tpu` as well as other [options](https://pytorch-
 
 If `gpu`, it can be helpful to specify the data parallelization strategy. This can be done using the `strategy` arg in `main.create_trainer()`
 
-```
+```python
 model.create_trainer(logger=comet_logger, strategy="ddp")
```
Comment (ethanwhite) [inline docs/user_guide/09_configuration_file.md:110]: Since this is a new section we should add the import here as well and switch from `model` to `m` for consistency

File: docs/user_guide/09_configuration_file.md [hunk 2]
```diff
@@ -144,13 +146,13 @@ Directory to search for images in the csv_file image_path column
 
 Learning rate for the training optimization. By default the optimizer is stochastic gradient descent with momentum. A learning rate scheduler is used based on validation loss
 
-```
+```python
 optim.SGD(self.model.parameters(), lr=self.config["train"]["lr"], momentum=0.9)
```
Comment (ethanwhite) [inline docs/user_guide/09_configuration_file.md:150]: Need to add `from torch import optim`?

File: docs/user_guide/09_configuration_file.md [hunk 3]
```diff
@@ -144,13 +146,13 @@ Directory to search for images in the csv_file image_path column
...
 ```
 
 A learning rate scheduler is used to adjust the learning rate based on validation loss. The default scheduler is ReduceLROnPlateau:
 
-```
+```python
 self.scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(self.optimizer, mode='min',
```
Comment (ethanwhite) [inline docs/user_guide/09_configuration_file.md:156]: Add `import torch`?

**Review by ethanwhite — COMMENTED** [review]
Really nice work @naxatra2! Three things to look at:

- In 01_Reading_data.md the changes related to `deepforest.utilities.read_file` have some issues. The initial changes don't work because the import `read_file` directly whereas the code that follows it runs `utilities.read_file` (i.e., the necessary import is `from deepforest import utilities`. Later changes switch to changing all code to importing `read_file` directly. The most important thing is that this should be consistent throughout. Since the PR is about the imports I lean towards using `from deepforest import utilities` throughout and leaving the calls to those functions untouched. This is consistent with submodule usage later in the docs.
- It looks like some similar changes should be added to `03_cropmodels.rst`
- I've made a bunch of inline comments related to other small fixes

File: docs/user_guide/11_training.md [hunk 4]
```diff
@@ -23,6 +23,10 @@ OSBS_029.jpg,161,155,199,191,Tree
 The config file specifies the path to the CSV file that we want to use when training. The images are located in the working directory by default, and a user can provide a path to a different image directory.
 
 ```python
+import os
+from deepforest import main
```
Comment (henrykironde) [inline docs/user_guide/11_training.md:27]: Are we using main?

**Review by henrykironde — CHANGES_REQUESTED** [review]
Thanks, @naxatra2,the changes. I've suggested a few additional places where we can make improvements.

**Comment by naxatra2** [comment]
I have noticed that in the docs `m` is not used consistently. It is sometimes used for `model.py`  and sometimes for `main.py`. For example we sometimes use `m = main.deepforest()` but we are also using `from deepforest import model as m`. Having both of these represented by a single letter makes it a bit hard to understand I guess ?

**Comment by ethanwhite** [comment]
Thanks @naxatra2 - this all looks good. Unfortunately I need to ask you to rebase one more time and it's going to be a little tricky due to #990 being merged. The trickiest piece is that 03_cropmodels was converted from rst to md. Sorry about that. The good news is that once this is done I think we're ready to merge!
